File: ios/Plugin/NoSleepPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(NoSleepPlugin, "NoSleep",
           CAP_PLUGIN_METHOD(stayAwake, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(sleep, CAPPluginReturnPromise);
)

File: ios/Plugin/NoSleepPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(NoSleepPlugin)
public class NoSleepPlugin: CAPPlugin {
    @objc func stayAwake(_ call: CAPPluginCall) {
      DispatchQueue.main.async {
          if(!UIApplication.shared.isIdleTimerDisabled) {
              UIApplication.shared.isIdleTimerDisabled = true
          }
        
          call.resolve()
      }
    }
  
  @objc func sleep(_ call: CAPPluginCall) {
      DispatchQueue.main.async {
          if(UIApplication.shared.isIdleTimerDisabled) {
              UIApplication.shared.isIdleTimerDisabled = false
          }
        
          call.resolve()
      }
  }
}
